src/zcash/address/mnemonic.h

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2021 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php .
+
+#ifndef ZCASH_ZCASH_ADDRESS_MNEMONIC_H
+#define ZCASH_ZCASH_ADDRESS_MNEMONIC_H
+
+#include "zip32.h"
+
+class MnemonicSeed: public HDSeed {
+private:
+    Language language;
+    SecureString mnemonic;
+
+    MnemonicSeed() {}
+
+    void SetSeedFromMnemonic() {
+        seed.resize(64);
+        zip339_phrase_to_seed(language, mnemonic.c_str(), (uint8_t (*)[64])seed.data());
```

Comment:
This returns a success/failure bool. Failure is reachable in the case of a mnemonic that is not valid UTF-8, for example.

Reply:
Issue opened: https://github.com/zcash/zcash/issues/5399